src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -1730,6 +1730,31 @@ public static StatementSyntax ParseStatement(string text, int offset = 0, ParseO
             }
         }
 
+        /// <summary>
+        /// Parse a MemberDeclarationSyntax. This includes all of the kinds of members that could occur in a struct declaration.
+        /// If nothing resembling a valid member declaration is found in the input, returns null.
+        /// </summary>
+        /// <param name="text">The text of the declaration.</param>
+        /// <param name="offset">Optional offset into text.</param>
+        /// <param name="options">The optional parse options to use. If no options are specified default options are
+        /// used.</param>
+        /// <param name="consumeFullText">True if extra tokens in the input following a declaration should be treated as an error</param>
+        public static MemberDeclarationSyntax ParseMemberDeclaration(string text, int offset = 0, ParseOptions options = null, bool consumeFullText = true)
+        {
+            using (var lexer = MakeLexer(text, offset, (CSharpParseOptions)options))
+            using (var parser = MakeParser(lexer))
+            {
+                var node = parser.ParseMemberDeclaration();
+                if (node == null)
+                {
+                    return null;
```

Comment:
can you actually ever get null back?  that seems very wonky given general roslyn parsing semantics (i.e., always succeeds, always consumes full input).

Reply:
The short answer is, yes, for a few reasons. First, if there is no input the caller typically (in a type declaration) doesn't *need* a member declaration. Second, any member declaration would require synthesizing a number of missing tokens, more than for any of the other contexts.